### ISSUE TITLE
published artifacts do not match branch/tag names

### ISSUE THREAD

**Issue by wfurt** [quictls/openssl#106]
this impacts anybody who wants to build from released artifacts instead of GitHub -  like package maintainers.
Consider this fragment
```
version=2.1.7
curl -O https://github.com/microsoft/msquic/archive/refs/tags/v${version}.tar.gz
tar xfz /v${version}.tar.gz
cd msquic-${version}
...
```

it works fine for other packet but not for quilts
```
version=openssl-3.0.7+quic1
https://github.com/quictls/openssl/archive/refs/tags/${version}.tar.gz
tar xfz /v${version}.tar.gz
cd openssl-${version} -> fails
```

if you look at the content, 
```
tar ftz openssl-3.0.7+quic1.tar.gz
openssl-openssl-3.0.7-quic1/
openssl-openssl-3.0.7-quic1/ACKNOWLEDGEMENTS.md
openssl-openssl-3.0.7-quic1/AUTHORS.md
openssl-openssl-3.0.7-quic1/CHANGES.md
...
```

Because the branch name has special character the files inside are renamed to avoid it. It does not matter for `git` as the name never appears on file system .

This obviously can be workaround but it would be really _nice_ if artifacts released from here follow standard convention. 

Since we will not change past tags, I suggest to use `-quic1` suffix instead of `+quic1`.

cc: @nibanks @manickap

**Comment by tmshort** [comment]
Hi, just in time for the next version.
I'll tag with -quic1 for these next releases.

**Comment by tmshort** [comment]
OK, I just released `openssl-3.0.8-quic1` and `OpenSSL-1_1_1t-quic1`.
